### DISCUSSION TITLE
Traditional Chinese not Supported

### DISCUSSION THREAD

**Discussion by danielbarenkamp** [mpdf/mpdf#1488] [Q&A]
### I found this bug

MPDF Does not seem to show traditional chinese characters:

Orginal Text:
<img width="293" alt="Original Text" src="https://user-images.githubusercontent.com/3169388/128645168-727381e5-15aa-4d5e-9980-57b139614e79.png">

<img width="198" alt="PDF Text" src="https://user-images.githubusercontent.com/3169388/128645169-06550971-b446-4755-8dd3-38e18b78891b.png">

### This is mPDF and PHP version and environment (server/fpm/cli etc) I am using
MPDF 8.0.10
PHP 7.3 (FPM)

### This is the PHP code snippet I use

```
<?php
$mpdf->useSubstitutions = TRUE;
$mpdf->autoScriptToLang = TRUE;
$mpdf->autoLangToFont = TRUE;
$mpdf->allow_charset_conversion = TRUE;
$mpdf->backupSubsFont = array(
  'dejavusanscondensed',
  'arialunicodems',
  'sun-exta',
);

```

### This is the HTML/CSS code snippet I use

```

```

**Comment by Klap-in** [comment]
Please note you left out a small as possible example with the mpdf call and a small piece of html you have used to replicate the case your self. Without example it is a guessing game for us. Thanks for the effort.

**Comment by danielbarenkamp** [comment]
Hey Garrit :) here is the code

```
		$defaultFontConfig = (new Mpdf\Config\FontVariables())->getDefaults();
		$fontData = $defaultFontConfig['fontdata'];

    	try {
			$mpdfConfig = array(
				'mode' => 'utf-8', 
				'format' => $format,    // format - A4, for example, default ''
				'default_font_size' => 0,     // font size - default 0
				'default_font' => '',    // default font family
				'margin_left' => 0,    	// 15 margin_left
				'margin_right' => 0,    	// 15 margin right
				'margin_top' => $headerMargin,     // 16 margin top
				'margin_bottom' => $footerMargin,    	// margin bottom
				'margin_header' => 0,     // 9 margin header
				'margin_footer' => 0,     // 9 margin footer
				'orientation' => 'P',  	// L - landscape, P - portrait
				'tempDir' => dirname( __FILE__ ) . '/../cache/',
				'fontDir' => array(
					plugin_dir_path( dirname( __FILE__ ) ) . 'vendor/mpdf/mpdf/ttfonts/',
					plugin_dir_path( dirname( __FILE__ ) ) . 'fonts/',
				),
			    'fontdata' => array_merge($fontData, $customFonts),
			);
			$mpdf = new \Mpdf\Mpdf($mpdfConfig);	

			$mpdf->useAdobeCJK = true;
			$mpdf->autoScriptToLang = true;
			$mpdf->autoLangToFont = true;

			$mpdf->backupSubsFont = array(
			  'dejavusanscondensed',
			  'arialunicodems',
			  'sun-exta',
			);

			$html = '送貨地點: *送貨地點: *送貨地點: * 地區地區地區';

			$mpdf->WriteHTML($css . $html);
			$mpdf->Output('test.pdf', 'I');
			die();
```

**Comment by finwe** [comment]
Your code example is not reproducible, it contains undefined variables and functions.

Anyway, works OK for me when I remove all lines causing errors.

![adobe cjk](https://user-images.githubusercontent.com/195675/128751104-138f0367-edbb-485a-bff9-f11902af38a0.png)

**Comment by danielbarenkamp** [comment]
Ehm sorry but I can not reproduce this. I even reinstalled MPDF now. Can you share your code? Or do I miss some installs locally or on my server?

**Comment by finwe** [comment]
My code is just this (cleaned up from unrelated code).

```php
<?php

require_once __DIR__ . '/bootstrap.php';

$mpdf = new \Mpdf\Mpdf([
	'mode' => 'utf-8',
]);

$mpdf->useAdobeCJK = true;

$html = '送貨地點: *送貨地點: *送貨地點: * 地區地區地區';

$mpdf->WriteHTML($html);
$mpdf->Output('', 'I');
```

Results in the screenshot above.

There must be some specific quirk in your setup/code/environment causing this, share more information.

**Reply by danielbarenkamp** [reply]
Hmm I copied that code on my server and it still displays broken :( 

Could it be because of this: https://mpdf.github.io/fonts-languages/cjk-languages.html ? I tried to follow the link but it does not work. You may got a new one?

**Reply by finwe** [reply]
Sure - https://helpx.adobe.com/acrobat/kb/windows-font-packs-32-bit-reader.html

Will update in the documentation

**Reply by danielbarenkamp** [reply]
Sorry but that also does not work ... is there any server spec that we need? I have the problem and one of my customer (who is chinese) has the same issue. Maybe a server language or PHP extension?

**Reply by finwe** [reply]
From the information you provided I simply don't know (and it is posibble I won't even with them).

**Comment by danielbarenkamp** [comment]
Sorry I have to ask another one - the other chinese chars are shown correctly, but these ones do not:

號 碼： {{id}}
<img width="596" alt="Bildschirmfoto 2022-01-17 um 22 15 57" src="https://user-images.githubusercontent.com/3169388/149837750-160e7a43-17ac-48f1-a183-f2fe7e543342.png">



```
require_once __DIR__ . '/bootstrap.php';

$mpdf = new \Mpdf\Mpdf([
	'mode' => 'utf-8',
]);

$mpdf->useAdobeCJK = true;

$html = '號 碼： {{id}}';

$mpdf->WriteHTML($html);
$mpdf->Output('', 'I');

```

Do you know why?
